src/Cpp/1-getting-started/1-HelloWindow/Input/Keyboard.cpp

```diff
@@ -0,0 +1,52 @@
+#include "Keyboard.hpp"
+#include "Input.hpp"
+
+#include <GLFW/glfw3.h>
+
+void Keyboard::HandleKeys(
+    GLFWwindow* window,
+    const std::int32_t key,
+    const std::int32_t scanCode,
+    const std::int32_t action,
+    const std::int32_t modifier)
+{
+    const auto input = static_cast<Input*>(glfwGetWindowUserPointer(window));
```

Comment:
prefer actual type over auto

Reply:
In this case, since the type already appears in the cast, I think we can avoid duplicating the type

Reply:
I don't disagree, though my personal habit is to use it for "long" types and if something is simple to type, especially without any namespaces, it usually qualifies as explicit for me rather than auto

Reply:
I agree that long types should be "auto'ed", but I believe that duplication can cause some errors, in case you would want to change the type for example, you would have to do it twice, instead of once, furthermore, although Input is very easy to type "auto" is shorter and still gets the point across thanks to the type being already very explicit in the cast.

Reply:
I am not 100% sure here, I am also in favour of that.

In this instance using explicit type